execution_engine/src/core/engine_state/mod.rs

```diff
@@ -158,42 +152,25 @@ where
             genesis_config_hash,
             protocol_version,
             correlation_id,
-            self.config,
+            *self.config(),
             ee_config.clone(),
             tracking_copy,
             system_module,
         );
 
-        // Create mint
-        let mint_hash = genesis_installer.create_mint()?;
+        genesis_installer.create_mint()?;
 
         // Create accounts
         genesis_installer.create_accounts()?;
 
         // Create handle payment
-        let handle_payment_hash = genesis_installer.create_handle_payment()?;
+        genesis_installer.create_handle_payment()?;
 
         // Create auction
-        let auction_hash = genesis_installer.create_auction()?;
+        genesis_installer.create_auction()?;
 
         // Create standard payment
-        let standard_payment_hash = genesis_installer.create_standard_payment();
-
-        // Associate given CostTable with given ProtocolVersion.
-        {
-            let protocol_data = ProtocolData::new(
-                *wasm_config,
-                *system_config,
-                mint_hash,
-                handle_payment_hash,
-                standard_payment_hash,
-                auction_hash,
-            );
-
-            self.state
```

Comment:
I tried your suggestion @mpapierski however, I run into an issue, earlier the genesis installer would handover it's instance of protocol data to setup a new runtime context for creating purses, however it then calls the host mint function which will then do a lookup for the contracts in global state, and it will error out if some default values haven't already been written in